brouter-map-creator/src/main/java/btools/mapcreator/ConvertLidarTile.java

```diff
@@ -186,4 +217,91 @@ public static void main(String[] args) throws Exception {
     doConvert(args[1], ilon_base, ilat_base, filename30);
   }
 
+  public SrtmRaster getRasterDirect(File f, double lon, double lat) throws Exception {
+    if (f.length() > ((SRTM3_ROW_LENGTH + 1) * (SRTM3_ROW_LENGTH + 1) * 2)) {
+      ROW_LENGTH = SRTM1_ROW_LENGTH;
+    } else {
+      ROW_LENGTH = SRTM3_ROW_LENGTH;
+    }
+    System.out.println("read file " + f + " rl " + ROW_LENGTH);
+
+    // stay a 1x1 raster
+    NROWS = ROW_LENGTH + 1;
+    NCOLS = ROW_LENGTH + 1;
+
+    imagePixels = new short[NROWS * NCOLS]; // 650 MB !
+
+    // prefill as NODATA
+    for (int row = 0; row < NROWS; row++) {
+      for (int col = 0; col < NCOLS; col++) {
+        imagePixels[row * NCOLS + col] = NODATA;
+      }
+    }
+
+    readHgtFromFile(f, 0, 0);
+
+    boolean halfCol5 = false; // no halfcol tiles in lidar data (?)
+
+    SrtmRaster raster = new SrtmRaster();
+    raster.nrows = NROWS;
+    raster.ncols = NCOLS;
+    raster.halfcol = halfCol5;
+    raster.noDataValue = NODATA;
+    raster.cellsize = 1. / (double) ROW_LENGTH;
+    raster.xllcorner = (int) (lon < 0 ? lon - 1 : lon); //onDegreeStart - raster.cellsize;
+    raster.yllcorner = (int) (lat < 0 ? lat - 1 : lat); //latDegreeStart - raster.cellsize;
+    raster.eval_array = imagePixels;
+
+    return raster;
+  }
+
+  public SrtmRaster getRasterZip(File f, double lon, double lat) throws Exception {
+
+    ZipInputStream zis = new ZipInputStream(new BufferedInputStream(new FileInputStream(f)));
+    try {
+      for (; ; ) {
+        ZipEntry ze = zis.getNextEntry();
+        if (ze.getName().toLowerCase().endsWith(".hgt")) {
+          if (ze.getSize() > ((SRTM3_ROW_LENGTH + 1) * (SRTM3_ROW_LENGTH + 1) * 2)) {
+            ROW_LENGTH = SRTM1_ROW_LENGTH;
+          } else {
+            ROW_LENGTH = SRTM3_ROW_LENGTH;
+          }
+          System.out.println("read file " + f + " rl " + ROW_LENGTH);
+          // stay a 1x1 raster
+          NROWS = ROW_LENGTH + 1;
+          NCOLS = ROW_LENGTH + 1;
+
+          imagePixels = new short[NROWS * NCOLS]; // 650 MB !
+
+          // prefill as NODATA
+          for (int row = 0; row < NROWS; row++) {
+            for (int col = 0; col < NCOLS; col++) {
+              imagePixels[row * NCOLS + col] = NODATA;
+            }
+          }
+          readHgtFromStream(zis, 0, 0, ROW_LENGTH);
+          break;
+        }
+      }
+    } finally {
+      zis.close();
+    }
+
+
+    boolean halfCol5 = false; // no halfcol tiles in lidar data (?)
+
+    SrtmRaster raster = new SrtmRaster();
+    raster.nrows = NROWS;
+    raster.ncols = NCOLS;
+    raster.halfcol = halfCol5;
+    raster.noDataValue = NODATA;
+    raster.cellsize = 1. / (double) ROW_LENGTH;
+    raster.xllcorner = (int) (lon < 0 ? lon - 1 : lon); //onDegreeStart - raster.cellsize;
+    raster.yllcorner = (int) (lat < 0 ? lat - 1 : lat); //latDegreeStart - raster.cellsize;
+    raster.eval_array = imagePixels;
+
+    return raster;
+
+  }
```

Comment:
Code duplication. Just write one method which reads from stream and call it either using a stream created from a file or from a zip.